src/client/Transport.ts

```diff
@@ -175,7 +179,20 @@ export class Transport {
             onconnect()
         };
         this.socket.onmessage = (event: MessageEvent) => {
-            onmessage(ServerMessageSchema.parse(JSON.parse(event.data)))
+            try {
+                const serverMsg = ServerMessageSchema.parse(JSON.parse(event.data));
+
+                // Handle validation errors
```

Comment:
We can remove this error handling code now that the server will handle invalid usernames.